modules/core/src/main/scala/org/tessellation/domain/trust/storage/OrdinalTrustMap.scala

```diff
@@ -0,0 +1,34 @@
+package org.tessellation.domain.trust.storage
+
+import cats.Order
+import cats.syntax.contravariant._
+
+import org.tessellation.cli.method.ordinalTrustUpdateInterval
+import org.tessellation.schema.SnapshotOrdinal
+
+import derevo.cats.eqv
+import derevo.circe.magnolia.encoder
+import derevo.derive
+
+@derive(eqv, encoder)
+case class OrdinalTrustMap(
+  ordinal: SnapshotOrdinal,
+  peerLabels: PublicTrustMap,
+  trust: TrustMap
+)
+
+object OrdinalTrustMap {
+
+  implicit val order: Order[OrdinalTrustMap] = Order[SnapshotOrdinal].contramap(_.ordinal)
+
+  def empty: OrdinalTrustMap = OrdinalTrustMap(SnapshotOrdinal.MinValue, PublicTrustMap.empty, TrustMap.empty)
+
+  def initial: OrdinalTrustMap = OrdinalTrustMap(SnapshotOrdinal(ordinalTrustUpdateInterval), PublicTrustMap.empty, TrustMap.empty)
+
+  def advanceInterval(trust: OrdinalTrustMap): OrdinalTrustMap = OrdinalTrustMap(
```

Comment:
Maybe I don't understand the usage here but should you discard `peerLabels` and `trust` fields and use the empty values for those instead of copy? If you really don't need those just take an ordinal as parameter - I would also adjust the name for the method if the functionality here is as intended.